### PR TITLE
Fix: Don't offer to create new items on read only collections

### LINKED ISSUE #32224: Don't offer to create new items on read only collections
### Describe the bug

We seem to be offering the "Create a new..." on collections which are read only
image

When the read-only clicks on those, they can't save in there
image

### To Reproduce

1) Create a collection
2) grant read only permission to collections to All Users
3) create a new normal (non-admin) user
4) go to the read-only collection, see the button to create

### Expected behavior

We shouldn't offer the user the possibility of creating a new item on a read-only collection

### Logs

NA

### Information about your Metabase installation

```JSON
47.x, but this comes from way before
```

### Severity

P3

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by JesseSDevaney**
**Problem:** Users with read-only access on empty collections still see an option to "Create a new..."

Context: https://github.com/metabase/metabase/issues/32224

---
**Desired Behavior**/**Acceptance Criteria**

_Writable, empty collection_

image

- [ ] Should **show** "Create a new..." button for writable, empty collections

_Read-only, empty collection_

image

- [ ] Should **not show** "Create a new..." button for read-only, empty collections

_Tests_

- [ ] **Writable:** Validated behavior for writable, empty collections
- [ ] **Read-only:** Validated behavior for read-only, empty collections

### CODE REVIEW

File: frontend/src/metabase/collections/components/CollectionEmptyState/CollectionEmptyState.tsx [hunk 1]
```diff
@@ -12,22 +12,26 @@ import {
...
 export interface CollectionEmptyStateProps {
-  collectionId?: CollectionId;
+  collection?: Collection;
 }
 
 const CollectionEmptyState = ({
-  collectionId,
+  collection,
 }: CollectionEmptyStateProps): JSX.Element => {
+  const canWrite = collection?.can_write ?? false;
```
Comment (iethree) [inline frontend/src/metabase/collections/components/CollectionEmptyState/CollectionEmptyState.tsx:21]: this is the most nitpicky code comment I have ever made

```suggestion
  const canWrite = !!collection?.can_write;
```

File: frontend/src/metabase/collections/components/CollectionEmptyState/CollectionEmptyState.unit.spec.tsx
Comment (iethree) [inline frontend/src/metabase/collections/components/CollectionEmptyState/CollectionEmptyState.unit.spec.tsx:1]: 👏  🏆  thanks for creating a new unit test spec for this! These are a big fat pain to set up and doing so makes it so much easier to add tests in the future. ❤️

File: e2e/test/scenarios/collections/permissions.cy.spec.js [hunk 2]
```diff
@@ -353,6 +353,29 @@ describe("collection permissions", () => {
...
+              });
+
+              // writable, empty collection
+              cy.findByTestId("main-navbar-root")
+                .findByText("Your personal collection")
+                .click();
+              cy.findByTestId("collection-empty-state")
+                .findByText("Create a new…")
+                .should("exist");
+            });
```
Comment (iethree) [inline e2e/test/scenarios/collections/permissions.cy.spec.js:377]: I think this is adequately unit tested, making this e2e test redundant. What do you think @nemanjaglumac ?

**Review by iethree — APPROVED** [review]
✨  only nitpick comments

File: e2e/test/scenarios/collections/permissions.cy.spec.js (hunk 2 above)
Comment (nemanjaglumac) [reply e2e/test/scenarios/collections/permissions.cy.spec.js:377]: 100%.
We can/should remove E2E tests in this case.

**Pull request merged**